services/FacilityService.php

```diff
@@ -0,0 +1,246 @@
+        $sql .= "     street='" . $data["street"] . "',";
+        $sql .= "     city='" . $data["city"] . "',";
+        $sql .= "     state='" . $data["state"] . "',";
+        $sql .= "     postal_code='" . $data["postal_code"] . "',";
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
+        $sql .= " WHERE id='" . $data["fid"] . "'";
+
+        return sqlStatement($sql);
+    }
+
+    public function insert($data)
+    {
+        $sql  = " INSERT INTO facility SET";
+        $sql .= "     name='" . $data["name"] . "',";
+        $sql .= "     phone='" . $data["phone"] . "',";
+        $sql .= "     fax='" . $data["fax"] . "',";
+        $sql .= "     street='" . $data["street"] . "',";
+        $sql .= "     city='" . $data["city"] . "',";
+        $sql .= "     state='" . $data["state"] . "',";
+        $sql .= "     postal_code='" . $data["postal_code"] . "',";
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
```

Comment:
Since it makes more sense to do sql  escaping here (rather than upstream since this can turn into an api), would surround every $data with the add_escape_custom() function.